app/Http/Controllers/Api/AuthController.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Models\User;
+
+use Validator;
+
+class AuthController extends Controller
+{
+    public function __construct() {
+        $this->middleware(['auth.basic.once'])->except(['register', 'login']);
+    }
+
+    public function register(Request $request) {
+        $validator = Validator::make($request->all(), [
+            'name' => 'required|min:2|max:64',
+            'email' => 'required|email|max:255',
+            'password' => 'required|confirmed|max:255'
+        ]);
+            
+        if ($validator->fails()) {
+            return response()->json($validator->errors()->toJson(), 400);
+        }
+
+        $role = strpos($request->email, 'stud') ? 1 : 2;
```

Comment:
just to be on the safe side, i'd go for `@stud` in case of some teacher having _stud_ in his email address